### ISSUE TITLE
Batch Size for Counterfactuals

### ISSUE THREAD

**Issue by HeyItsBethany3** [SeldonIO/alibi#625]
Hi team, 

What is the batch size used for within the counterfactual RL algorithm? Please could you explain the process in a bit more detail. 

Thank you!

**Comment by mauicv** [comment]
Hey @HeyItsBethany3,
Thanks for opening the issue.

Generally speaking, the `batch_size` is firstly the number of samples drawn from the dataset and stored in the replay buffer on each training iteration. This process involves sampling the instances, generating counterfactuals and computing the reward. All this data is stored in the replay buffer.

It’s also the number of samples we draw from the replay buffer which we use to train the actor and critic models. This is done in the typical machine learning sense. So we compute the loss over a batch of samples, compute the gradient and perform a gradient descent step.

Finally, once the model has been fit, If you pass a large number of instances to the explain method it batches them into sets of size `batch_size` and passes them to the model batch by batch. This is just to limit the number the model is handling at any one time.

The question is quite general, It would help if you could be a bit more specific? Is there Something specifically your trying to do or understand?

**Comment by HeyItsBethany3** [comment]
No that's really helpful, thank you!
